### PR TITLE
Maintenance Splash Page

### PR DESCRIPTION
**Pull request by DaedalusG**
This PR makes a maintenance splash page intended to display service health after installation completes. 
Intended as WIP no nav yet

Screenshot 2024-07-24 at 12 00 44 AM

## Test plan
Run local Golden Tests

### CODE REVIEW

File: internal/appliance/reconciler/reconcile.go [hunk 1]
```diff
@@ -79,62 +79,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
+			reconcile = r.reconcileGrafana
+		case "jaeger":
+			reconcile = r.reconcileJaeger
+		case "otel":
+			reconcile = r.reconcileOtel
+		}
+
+		if err := reconcile(ctx, &sourcegraph, &applianceSpec); err != nil {
+			return ctrl.Result{}, errors.Newf("failed to reconcile %s: %w", service, err)
+		}
```
[the remark this answers is not in the document]
Comment (DaedalusG) [reply internal/appliance/reconciler/reconcile.go:127]: Mines cooler though

[the remark this answers is not in the document]
Comment (DaedalusG) [reply internal/appliance/reconciler/reconcile.go:127]: okay Ill change it back

File: internal/appliance/json.go [hunk 2]
```diff
@@ -101,7 +102,7 @@ func (a *Appliance) getStatusJSONHandler() http.Handler {
 			Data:   "",
 		}
 
-		if err := a.writeJSON(w, http.StatusOK, responseData{"status": data}, nil); err != nil {
```
Comment (DaedalusG) [inline internal/appliance/json.go:104]: I added these to fix an error in CI from bazel:
```
ERROR: /Users/warrengifford/sourcegraph/internal/appliance/BUILD.bazel:4:11: GoCompilePkg internal/appliance/appliance.a failed: (Exit 1): builder failed: error executing GoCompilePkg command (from target //internal/appliance:appliance) bazel-out/darwin_arm64-opt-exec-ST-a828a81199fe/bin/external/go_sdk/builder_reset/builder compilepkg -sdk external/go_sdk -installsuffix darwin_arm64 -src internal/appliance/appliance.go -src ... (remaining 111 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: nogo: errors found by nogo during build-time code analysis:
internal/appliance/json.go:21:85: (*Appliance).writeJSON - headers always receives nil (unparam)
```
I think that three calls of the same func with nil for the param triggers this build failure

**Review by Chickensoupwithrice — APPROVED** [review]
kind of a rubber stamp, but seems like a reasonable change :)

**Pull request merged**